File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int start = 0;
        for (int i = 0; i < t.length()-p.length()+1; i++) {
            String num = "";
            for (int j = 0; j < p.length(); j++) {
                num += t.charAt(start+j);
            }
            if (Long.parseLong(num) <= Long.parseLong(p)) {
                answer++;
            }
            start++;
        }
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    lst = []
    for i in range(len(t)-len(p)+1):
        num = ""
        for j in range(len(p)):
            num += t[i+j]
        lst.append(int(num))
    for i in lst:
        if i <= int(p):
            answer += 1
    return answer